### PR TITLE
Travis: Test build against system libs (& Qt4)

### PR DESCRIPTION
**Pull request by luke-jr**
Also removes now-unnecessary `bc` dependency from other builds.

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK on testing with qt4, but I'm not entirely happy about adding yet another build for this, Travis is slow enough these days.

**Comment by jonasschnelli** [comment]
Concept ACK.
Agree with @laanw but IMO another build is more valuable then a speedy CI.

**Comment by laanwj** [comment]
@jonasschnelli I wonder if there's a compromise, if we can combine the qt4 build with one of the existing builds

**Comment by jonasschnelli** [comment]
We only have one linux "full" Qt5 build "32bit + dash". The other linux QT5 build does not build the wallet. Maybe it would be worth to change one of the windows builds (Win32) to build with Qt4? On the other hand, Qt4 mostly makes sense on Linux, but even though, a Win32 Qt4 build should at least detect most Qt4 incompatibilities.

**Comment by laanwj** [comment]
Yeah. That may work.
Anyhow this is up to @theuni. Another option would be to combine this build with something else that we wanted to test already, so that it's not just added for qt4.

**Comment by luke-jr** [comment]
@theuni Can't seem to get this to build with Qt - any ideas? Shouldn't --with-qt=qt4 configure *fail* if it has a problem? :/

**Comment by luke-jr** [comment]
@laanwj Oh, an idea if Travis time is a concern: make this build use *only* system libraries (no depends). That's probably better test coverage anyway...

**Comment by laanwj** [comment]
> @laanwj Oh, an idea if Travis time is a concern: make this build use only system libraries (no depends). That's probably better test coverage anyway...

Not a bad idea because of the extra test coverage with old libraries, but I don't think that will save much time as the depends are supposed to be built once and cached anyway.

**Comment by MarcoFalke** [comment]
> supposed to be built once and cached anyway.

Does travis cache between builds, I don't think so?

**Comment by laanwj** [comment]
No, not between builds, I think

**Comment by theuni** [comment]
Right. To clarify, depends are cached and re-used for subsequent builds of the same config. Each config is an island though.

Personally, I don't think this is worth adding another build for. If anything, I'd rather do as @laanwj said and combine it with another.

**Comment by laanwj** [comment]
Travis PR test is quite slow lately. I'd certainly prefer not to add another config.

Testing more obscure platforms and library versions (like Qt4) may be better suited for something like a nightly extended build (e.g. #7148)

**Comment by luke-jr** [comment]
Once #7339 is merged, I suggest this can be extended to test --without-libevent as well.

**Comment by laanwj** [comment]
Closing, this is part of #7148 now

**Comment by laanwj** [comment]
Needs rebase.

**Comment by jtimon** [comment]
Concept ACK

**Comment by luke-jr** [comment]
Rebased, sorry for the delay

**Comment by TheBlueMatt** [comment]
Concept ACK. Only concern is are we willing to add more jobs to travis? Can we pay them to make things go faster?

**Comment by MarcoFalke** [comment]
@TheBlueMatt According to apache, it is possible to increase the number of concurrent builds. (https://blogs.apache.org/infra/entry/apache_gains_additional_travis_ci)

Though, I am slightly worried that this makes it harder for developers to fork the project and enable travis on their fork. After switching to the paid infrastructure, we might have faster machines or even extended timeouts for the build times. By forking the project, the new limits no longer apply and builds might time out or fail, which is a nuisance for new contributors.

**Comment by MarcoFalke** [comment]
Is this still WIP or ready for review?

File: .travis.yml [hunk 1]
```diff
@@ -25,6 +25,8 @@ env:
     - HOST=arm-linux-gnueabihf PACKAGES="g++-arm-linux-gnueabihf" DEP_OPTS="NO_QT=1" CHECK_DOC=1 GOAL="install" BITCOIN_CONFIG="--enable-glibc-back-compat --enable-reduce-exports"
 # Win32
     - HOST=i686-w64-mingw32 DPKG_ADD_ARCH="i386" DEP_OPTS="NO_QT=1" PACKAGES="python3 nsis g++-mingw-w64-i686 wine1.6 bc" RUN_TESTS=true GOAL="install" BITCOIN_CONFIG="--enable-reduce-exports"
+# Qt4
+    - HOST=x86_64-unknown-linux-gnu PACKAGES="bc python3-zmq qt4-dev-tools" DEP_OPTS="NO_QT=1" RUN_TESTS=true GOAL="install" BITCOIN_CONFIG="--enable-zmq --enable-glibc-back-compat --enable-reduce-exports --with-qt=qt4 CPPFLAGS=-DDEBUG_LOCKORDER"
```
Comment (MarcoFalke) [inline .travis.yml]: No need for the `bc` package. Also, the configure option is called `--with-gui=qt4`.

Comment (luke-jr) [inline .travis.yml]: Why is `bc` not needed here? Other builds need it...?

Comment (MarcoFalke) [inline .travis.yml]: I can't figure out why the "GNU bc arbitrary precision calculator language" is required. It  is likely an artifact of the days where the functional tests were written in bash.

Comment (MarcoFalke) [inline .travis.yml]: https://github.com/bitcoin/bitcoin/blob/2191eac812a72fe57167be882bddd2e6a3f0340e/qa/rpc-tests/util.sh#L38

**Comment by luke-jr** [comment]
Fixed issues, rebased, and removed `bc` from the other builds too.

**Comment by luke-jr** [comment]
There, tests pass now, and it tests against more system dependencies as well.

**Comment by MarcoFalke** [comment]
utACK 6d2aac8bb142adc1e5cff9dd91ddd1d91403b33c

**Pull request merged**